Doc/library/threading.rst

```diff
@@ -264,7 +264,8 @@ since it is impossible to detect the termination of alien threads.
    *target* is the callable object to be invoked by the :meth:`run` method.
    Defaults to ``None``, meaning nothing is called.
 
-   *name* is the thread name.  By default, a unique name is constructed of the
+   *name* is the thread name. If *target* is specific but *name* is omitted,
+   use ``target.__name__``. By default, a unique name is constructed of the
```

Comment:
This now is slightly confusing. What about 

> In any other case, a unique name is...

Reply:
I applied your proposed change.